src/radiation_system.hpp

```diff
@@ -687,6 +693,108 @@ AMREX_GPU_DEVICE auto RadSystem<problem_t>::ComputeCellOpticalDepth(const quokka
 						       // return 0.5*(tau_L + tau_R); // arithmetic mean
 }
 
+template <typename problem_t>
+template <FluxDir DIR, typename TARRAY>
```

Comment:
Does TARRAY actually need to be a template parameter? It looks like you call this function with the same types every time.

Reply:
It was used as std array and as valarray in two different places.

Reply:
I didn't some changes to the definition and calling of ComputeRadPressure. Now the syntax if slightly more complex but clearer.